org.eclipse.ice.dev/org.eclipse.ice.dev.annotations/src/main/java/org/eclipse/ice/dev/annotations/processors/ICEAnnotationExtractionService.java

```diff
@@ -55,10 +93,20 @@ public ICEAnnotationExtractionService(Elements elementUtils, ObjectMapper mapper
 		this.nameGenerator = nameGenerator;
 	}
 	
```

Comment:
My concerns lie more in that fact that the initialization of these instances are necessary for the class to function as a whole.  I could assign them each an instance of a default implementation(allowing for setters to customize these facets), otherwise the user is liable to use the no args constructor, call extract, and get a null pointer exception.

Reply:
Though I cant do this for elementUtils and processingEnv, as they are essential but cannot be instantiated properly in the scope of the class, so forcing the client to supply these upon instantiation is the only way to ensure safe use of the class.